integrations/access/datadog/bot.go

```diff
@@ -32,9 +33,13 @@ import (
 	"github.com/gravitational/teleport/integrations/access/accessrequest"
 	"github.com/gravitational/teleport/integrations/access/common"
 	"github.com/gravitational/teleport/integrations/lib"
+	"github.com/gravitational/teleport/integrations/lib/logger"
 	pd "github.com/gravitational/teleport/integrations/lib/plugindata"
 )
 
+// AutoApprovalsAnnotation defines the datadog auto approvals label.
+const AutoApprovalsAnnotation = "datadog_auto_approvals"
```

Comment:
With new plugins we tried to introduce new standard annotations (with very moderate success) following the pattern `teleport.dev/something`.

Existing annotations are:
- `teleport.dev/notify-services` to choose who we should notify (deprecated, AMRs are a better way to do this)
- `teleport.dev/schedules` to specify the on-call schedule names for auto-approval
- `teleport.dev/teams` to choose which team we should notify (yes this makes absolutely no sense, this should be `notify-teams`)

The concepts of schedules and teams might not make sense for every service provider, we can either:
- introduce another annotation (but the `teams` annotation is now coined 😫 )
- reuse an existing one (only the `teleport.dev/schedules` does auto-approval today)

Those constants are stored there: https://github.com/gravitational/teleport/blob/4b3fe19d972391152d56b1744efd4bcfa32956dd/api/types/constants.go#L792-L797

Reply:
After some more thought, is there any concern with this annotation being reused for multiple plugins? I could see a user unintentionally adding automatic approvals for another plugin's service/team with the same names.

For example, if I have a Datadog team that has the same name as an PagerDuty service, then there is no option to enable automatic approvals for only the Datadog team.

Reply:
> After some more thought, is there any concern with this annotation being reused for multiple plugins? I could see a user unintentionally adding automatic approvals for another plugin's service/team with the same names.

I agree that this is a questionable design but this is what was done for the last plugins. We should get rid of annotations entirely by moving approval via AMRs as well. In the meantime I'd recommend to stick to the current broken plan rather than changing again. This is not a blocking comment, you might want to check with Roman before merging.

Reply:
That sounds reasonable. I'm okay with sticking to the current annotation pattern for now and improving the workflow with AMRs in the future.